Tests/SourceKittenFrameworkTests/ModuleTests.swift

```diff
@@ -22,24 +22,24 @@ class ModuleTests: XCTestCase {
         let temporaryURL = URL(fileURLWithPath: NSTemporaryDirectory())
             .appendingPathComponent("\(#function)-\(NSUUID())")
         try FileManager.default.createDirectory(at: temporaryURL, withIntermediateDirectories: true)
-        let cloneArguments = ["git", "clone", "https://github.com/Carthage/Commandant.git"]
-        let cloneResult = Exec.run("/usr/bin/env", cloneArguments, currentDirectory: temporaryURL.path)
+        let cloneArguments = ["clone", "https://github.com/Carthage/Commandant.git"]
+        let cloneResult = Exec.run("C:\\Program Files\\Git\\cmd\\git.exe", cloneArguments, currentDirectory: temporaryURL.path)
```

Comment:
Can you introduce new functions in `Exec.swift` that abstracts platform-specific execution of the commands we use in SourceKitten? For example:

```swift
extension Exec {
    static func runGit(_ arguments: String...,
                       currentDirectory: String = FileManager.default.currentDirectoryPath,
                       stderr: Stderr = .inherit) -> Results {
#if os(Windows)
        let command = #"C:\Program Files\Git\cmd\git.exe"#
#else
        let command = "/usr/bin/env"
        let arguments = ["git"] + arguments
#endif
        return run(command, arguments, currentDirectory: currentDirectory, stderr: stderr)
    }
}
```

Reply:
I like the idea of the helper.  However, I do wonder if we should replicate `/usr/bin/env`?  I think that we want that for Windows as well, but does not have that.  We could explicitly use `CreateProcess` but that is a lot of complexity.

Reply:
The equivalent of `/usr/bin/env` on windows would certainly be helpful.

Maybe some abstraction should live in https://github.com/apple/swift-system? In any case I’m happy to consider it out of scope for initial Windows support in SourceKitten.

Reply:
I put together https://github.com/compnerd/Finder and we can simply do `which("git")` and `which("swiftc")` - it respects `PATHEXT` which means that we don't even need to worry about the suffix.